src/dc_tools.rs

```diff
@@ -242,9 +243,9 @@ pub fn dc_get_filesuffix_lc(path_filename: impl AsRef<str>) -> Option<String> {
 
 /// Returns the `(width, height)` of the given image buffer.
 pub fn dc_get_filemeta(buf: &[u8]) -> Result<(u32, u32), Error> {
-    let meta = image_meta::load_from_buf(buf)?;
-
-    Ok((meta.dimensions.width, meta.dimensions.height))
+    let image = image::io::Reader::new(Cursor::new(buf));
+    let dimensions = image.into_dimensions()?;
```

Comment:
this does not create the whole image object (all data may take several mb uncompressed) but only reads dimensions?

Reply:
Yes, it does not call the decoder, at least according to documentation and for JPEG according to the code.

Reply:
See https://docs.rs/image/0.23.7/image/io/struct.Reader.html#method.into_dimensions

`decode` function is right below

Reply:
great improvement then, thanks a lot!